### PR TITLE
Keep parsed qsos

### PR DESCRIPTION
**Pull request by dl1jbe**
Keep 'qso_array' as parsed representation of the log in memory. 

Keep array up to date while adding or editing new contacts.
Each entry keeps also a copy of the original log line. so we can move away from using the fixed size 'qsos[]' array.

### CODE REVIEW

**Comment by dl1jbe** [comment]
All done. Should be ready for merge now.

File: src/last10.c [hunk 1]
```diff
@@ -28,49 +28,44 @@
 #include "get_time.h"
 #include "globalvars.h"		// Includes glib.h and tlf.h
 
-
 int last10(void) {
```
Comment (zcsahok) [inline src/last10.c:31]: Not related to this PR: `last10()` considers only the hour+minute,  +/- 1 day. So opening an older log file sometimes shows last10 as if it they would be recent QSOs.

File: src/searchlog.c [hunk 2]
```diff
@@ -373,37 +375,39 @@ int bandstr2line(char *buffer) {
...
 	}
 
-	g_strlcpy(s_inputbuffer, qsos[qso_index] + 29, 13); /* call */
-	if (strstr(s_inputbuffer, hiscall) == 0) {
+	if (strstr(qso->call, call) == 0) {
 	    continue;   // no match
 	}
 
-	g_strlcpy(searchresult[srch_index], qsos[qso_index], 81);
+	g_strlcpy(searchresult[srch_index], QSOS(qso_index), 81);
```
Comment (zcsahok) [inline src/searchlog.c:410]: Just for the TODO list: rework searching to use parsed QSO data. Now it practically parses log line again in `extractData()`.

**Review by zcsahok — APPROVED** [review]
Fine with me. Just added 2 notes for future.

**Comment by dl1jbe** [comment]
QSOS ist used in the following places:

1. As source for edit_last() function
2. Two times in genqtclist.c (one of it can use qso->call)
3. In write_qtclog_fm_cabr() which also can use the already parsed members
4. Rewriting the log in readcalls()
5. In scroll_log() for display of log lines and finally
6. above mentioned searchlog() function.

For 1., 4. and 5. we need the log line as text which could be generated from the parsed data. With that we could drop storing the line but would need to regenerate it later. Of that 5. is used quite often.

The other three places should be converted to using the parsed data.

File: src/last10.c (hunk 1 above)
Comment (dl1jbe) [reply src/last10.c:31]: Yes, adding the date to the calculation would be better.

**Pull request merged**